Compile switch for different handheld contact switches & record greeting on the phone itself & extensive documentation

DD4WH opened the pull request.

DD4WH committed:
Put all building info into one place in the readme

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Added compile switch for different handheld switches

DD4WH committed:
Update README.md

DD4WH committed:
Merge branch 'playfultechnology:main' into main

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
added documentation in ino file

DD4WH committed:
added photo

DD4WH committed:
Update README.md

DD4WH committed:
photo of W61 build

DD4WH committed:
added telephone model info

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

robinvan0418 commented on audio-guestbook.ino:
```diff
@@ -38,6 +44,10 @@
 #define HOOK_PIN 0
 #define PLAYBACK_BUTTON_PIN 1
 
+// comment his out, if your handheld OPENS the contact on lift
```
Found a typo here :)

DD4WH replied:
Thanks!
Corrected and added a lot more configuration options and made the sketch much easier to use for the beginner.

DD4WH committed:
greeting message can be recorded on the telephone

* code checks whether greeting message is present
* if not, it prompts the user to record the greeting message and saves the recording as "greeting.wav"
* if yes, it plays the greeting message as usual
* if you want to record a new greeting message, just delete the old one :-)

DD4WH committed:
Update README.md

DD4WH committed:
Added structured user configuration possibility in the sketch 

the user can configure three pre-compile settings by simply uncommenting/commenting #defines in the sketch:
1.) does your handheld OPEN or CLOSE the switch when you lift the handheld ?
2.) do you want the telephone to be able to record the greeting message automatically, if there is no "greeting.wav" present on the SD card ?
3.) do you use Teensy 4.0 (SD card slot of the audio board) or do you use Teensy 4.1 (SD card slot of the Teensy 4.1) ?

DD4WH committed:
More detailed info on configuration options

DD4WH committed:
optimized order of compile instructions

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Hardware photo added

DD4WH committed:
more information on building the hardware

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
added hardware needed

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

Brototype commented on audio-guestbook.ino:
```diff
@@ -27,16 +34,46 @@
-// DEFINES
 // Define pins used by Teensy Audio Shield
+#ifdef TEENSY_41
+#define SDCARD_CS_PIN    BUILTIN_SDCARD
```
if I use a teensy 4.1 with the same sd card I get a very "jumpy" recording. it skips audio pieces every couple of seconds. if is comment that line and put the sd card into the audio shield it works fine. can you help me to understand what might happen here?

Brototype reviewed (commented):
thanks a lot for this - I tried it out but I have problems with recording to the sd card straight out of the teensy 4.1

DD4WH committed:
Update README.md

alastaira commented:
Hi, thanks for this.

While I agree it's in some ways convenient, my reservation about being able to record the greeting on the device itself is that it is possible for guests to accidentally (or deliberately!) over-record the greeting, and I can't really think of a scenario where you'd need to change the greeting "in the field" - can't it just be something recorded in advance and uploaded when the phone is built?

DD4WH commented:
> Hi, thanks for this.
> 
> While I agree it's in some ways convenient, my reservation about being able to record the greeting on the device itself is that it is possible for guests to accidentally (or deliberately!) over-record the greeting, and I can't really think of a scenario where you'd need to change the greeting "in the field" - can't it just be something recorded in advance and uploaded when the phone is built?

No, thats a misunderstanding of the function. You can only record a greeting message, if there is no greeting.wav on the SD card. So, no accidental or deliberate overwrite of the greeting message is possible, once a greeting message is recorded or a prerecorded greeting.wav is present on the SD card. 

The primary intention of building this was motivated by the observation that MANY builders were unable to realize that they had to record a greeting message in 44.1ksps sample rate. Every greeting message recorded in other sample rates would be refused to be played back on the Teensy. So I decided to add the function to record the greeting message on the Teensy itself in order to ensure that the greeting.wav would then be playable by the Teensy.

jvallano commented:
> > Hi, thanks for this.
> > While I agree it's in some ways convenient, my reservation about being able to record the greeting on the device itself is that it is possible for guests to accidentally (or deliberately!) over-record the greeting, and I can't really think of a scenario where you'd need to change the greeting "in the field" - can't it just be something recorded in advance and uploaded when the phone is built?
> 
> No, thats a misunderstanding of the function. You can only record a greeting message, if there is no greeting.wav on the SD card. So, no accidental or deliberate overwrite of the greeting message is possible, once a greeting message is recorded or a prerecorded greeting.wav is present on the SD card.
> 
> The primary intention of building this was motivated by the observation that MANY builders were unable to realize that they had to record a greeting message in 44.1ksps sample rate. Every greeting message recorded in other sample rates would be refused to be played back on the Teensy. So I decided to add the function to record the greeting message on the Teensy itself in order to ensure that the greeting.wav would then be playable by the Teensy.

Would there be a way to initiate a re-record of the greeting if the user messes up or stumbles during this process?

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md

DD4WH committed:
Update README.md
